### DISCUSSION TITLE
The Power: Scripts for Comprehensive Automated API Testing

### DISCUSSION THREAD

**Discussion by loujr** [community/community#61287] [Discover: GitHub Best Practices]
### Welcome to the [The Power](https://github.com/gm3dmo/the-power#the-power)

The Power is a third-party API tool integrating hundreds of shell scripts to provide plug and play options for building and testing GitHub's API.

Chances are that the Power has a script for everything you need to do, as well as many things that you may not have even known were possible! Using The Power you can:

- Save time with **hundreds of pre-tested, battle-hardened scripts** 
- Test automation workflows **without risking Production**
- **Spin up testing environments** fast
- **Run API diagnostic tests** fast

#### Feel The Power

The Power reduces GitHub's complexity to mere seconds. Don't waste time debugging - set up and throw away test environments as fast as you need to. Your well-weaponed DevOps team can adhere to the industry standard of "cattle not pets".

- Power through setup and get to coding
- Spin up resources for fast, replicable testing
- Build uncommon tests like `CODEOWNERS` files
- Replicate testing for multiple repositories

#### Trust The Power

Every endpoint is thoroughly tested before being added into The Power's framework. Of course, API endpoints might change over time, but every script is verified functional against the version of the API it was written against. 

- Basic, easily readable, and understandable
- Built to spec with GitHub's API versioning
- Flex to fit a wide variety of different use-cases

#### Wield The Power

- Take a peak at the [README](https://github.com/gm3dmo/the-power/blob/main/docs/README.md)
- [Get Started with Setup](https://github.com/gm3dmo/the-power/blob/main/docs/setup.md)
- Be aware of [Known Issues](https://github.com/gm3dmo/the-power/blob/main/docs/known-issues.md)

#### Can you handle The Power?

The Power is intended for developers with significant experience working with API endpoints programmatically and/or shell scripts. If this tool doesn't meet your needs, you can learn more from GitHub's REST API [public documentation](https://docs.github.com/en/rest?apiVersion=2022-11-28) including a [list of available endpoints](https://docs.github.com/en/rest/overview/endpoints-available-for-github-app-installation-access-tokens?apiVersion=2022-11-28).

<div class="flash mt-3 flash-warn flash-messages">
  
  <svg class="octicon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M8.22 1.754a.25.25 0 00-.44 0L1.698 13.132a.25.25 0 00.22.368h12.164a.25.25 0 00.22-.368L8.22 1.754zm-1.763-.707c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0114.082 15H1.918a1.75 1.75 0 01-1.543-2.575L6.457 1.047zM9 11a1 1 0 11-2 0 1 1 0 012 0zm-.25-5.25a.75.75 0 00-1.5 0v2.5a.75.75 0 001.5 0v-2.5z"></path></svg>

As The Power is a third-party application, it is not supported by GitHub support. Please exercise caution, as any script may cause unexpected behavior in an organization or repository, including potentially destructive actions.

</div>

<div class="flash flash-messages">
  
  <svg class="octicon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm6.5-.25A.75.75 0 017.25 7h1a.75.75 0 01.75.75v2.75h.25a.75.75 0 010 1.5h-2a.75.75 0 010-1.5h.25v-2h-.25a.75.75 0 01-.75-.75zM8 6a1 1 0 100-2 1 1 0 000 2z"></path></svg>
Please be aware that The Power is a passion project. This is a useful and unique tool but should not be considered complete and production ready. Please <b>do not</b> spam or harass the developer.
</div>

**Comment by chat14** [comment]
Facebook group me
https://facebook.com/groups/853224305713321/

**Comment by Smi41** [comment]
Hi

**Comment by Mariamasanenianthio1310** [comment]
Hello, I'm a student,  I don't know if this is the right place to ask my question but I'm trying my luck.

I work on a project which consists in inserting my own documents (pdf) in azure openAI and from GPT 3.5 Turbo, I deploy a web application, in which I ask my questions in the assistant and the answers will be based on of the document.

Only I can not savor the distinction between the standard model and the personalized model, because it seems to me they do the same thing.

What I understood is that the custom model if it does not find information in our imported data, it will not be able to look for additional information in the pretrainer model of GTP3 unlike the standard model. Am I wrong or not?

PS: I have already read the documents in the Microsoft site.
